content/kubernetes/re-clusters/multi-namespace.md

```diff
@@ -0,0 +1,176 @@
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["update", "get", "read", "list", "listallnamespaces", "watch", "watchlist",
+            "watchlistallnamespaces", "create","patch","replace","delete","deletecollection"]
+  - apiGroups: [""]
+    resources: ["endpoints"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: [""]
+    resources: ["events"]
+    verbs: ["create"]
+  - apiGroups: [""]
+    resources: ["services"]
+    verbs: ["get", "watch", "list", "update", "patch", "create", "delete"]
+```
+
+`role_binding.yaml` example:
+
+```yaml
+ kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: redb-role
+  labels:
+    app: redis-enterprise
+subjects:
+- kind: ServiceAccount
+  name: redis-enterprise-operator
+  namespace: <rec-namespace>
+- kind: ServiceAccount
+  name: redis-enterprise-admission
+  namespace: <rec-namespace>
+- kind: ServiceAccount
+  name: <service-account-name>
+  namespace: rec-namespace
+roleRef:
+  kind: Role
+  name: redb-role
+  apiGroup: rbac.authorization.k8s.io
+```
+
+Apply the files:
+
+```sh
+kubectl apply -f role.yaml
+kubectl apply -f role_binding.yaml
+```
+
+## Update Redis Enterprise operator ConfigMap
+
+There are two ways to update the operator ConfigMap (`operator-environment-config`) to specify which namespaces to manage.
+
+This ConfigMap can be created manually before deploying the RedisEnterpriseCluster, or will be created automatically when a Redis Enterprise cluster (REC) is deployed.
+
+{{<note>}}
+If the REC is configured to watch a namespace without setting the role and role binding permissions, or a namespace that's not created yet, the operator will fail and halt normal operations.
+{{</note>}}
+
+### Method 1: Namespace label (recommended)
+
+1. Create the `cluster_role_binding.yaml` and `cluster_role.yaml` files to add the namespace of the operator. Replace the `<rec-namespace>` with the namespace the Redis Enterprise cluster (REC) resides in.
+
+  `cluster_role.yaml` example:
+
+  ```yaml
+    apiVersion: rbac.authorization.k8s.io/v1
+    kind: ClusterRole
+    metadata:
+      name: redis-enterprise-operator
+    rules:
+      - apiGroups: [""]
+        resources: ["namespaces"]
+        verbs: ["list", "watch"]
+  ```
+
+  `cluster_role_binding.yaml` example: 
+
+  ```yaml
+    kind: ClusterRoleBinding
+    apiVersion: rbac.authorization.k8s.io/v1
+    metadata:
+      name: redis-enterprise-operator
+    subjects:
+    - kind: ServiceAccount
+      name: redis-enterprise-operator
+      namespace: <rec-namespace>
+    roleRef:
+      kind: ClusterRole
+      name: redis-enterprise-operator
+      apiGroup: rbac.authorization.k8s.io
+  ```
+
+2. Apply the files:
+
+  ```sh
+  kubectl apply -f cluster_role.yaml
+  kubectl apply -f cluster_role_binding.yaml 
+  ```
+
+3. Patch the ConfigMap in the REC namespace (`<rec-namespace>`) to identify the managed namespaces with your desired label (`<namespace-label>`).
```

Comment:
We should probably mention the two syntax options for REDB_NAMESPACES_LABEL:
1. It could be a `REDB_NAMESPACES_LABEL: name=value` in which both the label name and value should match.
2. Or `REDB_NAMESPACES_LABEL: name`, in which only the label name should match, and the value is disregarded.

Then, in the kubectl example below, we can say:
```sh
kubectl label namespace <managed-namespace> <label-name>=<label-value>
```

Whereas <label-value> should match the configuration for option 1, or can be anything for option 2.

Reply:
My previous explanation for option (2) was incorrect.
The correct semantics is that if only a label name is provided (`REDB_NAMESPACES_LABEL: name`), then the label value needs to be empty, as in:
```
kubectl label namespace <managed-namespace> <label-name>=""
```
or
```
kubectl label namespace <managed-namespace> <label-name>=
```